mgmtfn/mesosplugin/cniserver.go

```diff
@@ -0,0 +1,220 @@
+	cniLog.Infof("parsed tenant name: %s", cniReq.pluginArgs.Labels.TenantName)
+	cniLog.Infof("parsed network name: %s", cniReq.pluginArgs.Labels.NetworkName)
+	cniLog.Infof("parsed network group: %s", cniReq.pluginArgs.Labels.NetworkGroup)
+
+	// set defaults
+	cniReq.endPointLabels = map[string]string{cniapi.LABEL_NETWORK_NAME: "default-net",
+		cniapi.LABEL_TENANT_NAME: "default"}
+
+	for _, label := range []string{cniapi.LABEL_TENANT_NAME, cniapi.LABEL_NETWORK_NAME,
+		cniapi.LABEL_NETWORK_GROUP} {
+
+		switch label {
+		case cniapi.LABEL_TENANT_NAME:
+			if len(cniReq.pluginArgs.Labels.TenantName) > 0 {
+				cniReq.endPointLabels[label] = cniReq.pluginArgs.Labels.TenantName
+			}
+			cniLog.Infof("netplugin label %s = %s", cniapi.LABEL_TENANT_NAME,
+				cniReq.endPointLabels[label])
+
+		case cniapi.LABEL_NETWORK_NAME:
+			if len(cniReq.pluginArgs.Labels.NetworkName) > 0 {
+				cniReq.endPointLabels[label] = cniReq.pluginArgs.Labels.NetworkName
+			}
+			cniLog.Infof("netplugin label %s = %s", cniapi.LABEL_NETWORK_NAME,
+				cniReq.endPointLabels[label])
+
+		case cniapi.LABEL_NETWORK_GROUP:
+			if len(cniReq.pluginArgs.Labels.NetworkGroup) > 0 {
+				cniReq.endPointLabels[label] = cniReq.pluginArgs.Labels.NetworkGroup
+			}
+			cniLog.Infof("netplugin label %s = %s", cniapi.LABEL_NETWORK_GROUP,
+				cniReq.endPointLabels[label])
+		}
+	}
+
+	cniReq.networkID = cniReq.endPointLabels[cniapi.LABEL_NETWORK_NAME] + "." +
+		cniReq.endPointLabels[cniapi.LABEL_TENANT_NAME]
+	cniLog.Infof("network fdn  %s", cniReq.networkID)
+	cniReq.endpointID = cniReq.networkID + "-" +
+		cniReq.pluginArgs.CniContainerid
+	cniLog.Infof("endpoint fdn %s", cniReq.endpointID)
+	return nil
+}
+
+func mesosNwIntfAdd(httpBody []byte) ([]byte, error) {
+	addReq := cniServer{}
+
+	cniLog.Infof("process intf add req ")
+
+	if err := addReq.parseCniArgs(httpBody); err != nil {
+		return createCniErrorResponse(err)
+	}
+
+	if err := addReq.createCniEndPoint(); err != nil {
+		return createCniErrorResponse(err)
+	}
+
+	// success
+	return addReq.createCniSuccessResponse()
+}
+
+func mesosNwIntfDel(httpBody []byte) ([]byte, error) {
+	retSuccess := []byte("")
+	delReq := cniServer{}
+
+	cniLog.Infof("process intf delete req")
+
+	if err := delReq.parseCniArgs(httpBody); err != nil {
+		return createCniErrorResponse(err)
+	}
+
+	if err := delReq.deleteCniEndPoint(); err != nil {
+		return createCniErrorResponse(err)
+	}
+
+	return retSuccess, nil
+}
+
+func unknownReq(w http.ResponseWriter, r *http.Request) {
+	cniLog.Infof("unknown http request at %q", r.URL.Path)
+	content, _ := ioutil.ReadAll(r.Body)
+	cniLog.Debugf("received http body content: %s", string(content))
+	w.WriteHeader(http.StatusServiceUnavailable)
+}
+
+// InitPlugin registers REST endpoints to handle requests from Mesos CNI plugins
+func InitPlugin(netPlugin *plugin.NetPlugin) {
+
+	cniLog = log.WithField("plugin", "mesos")
+	cniLog.Infof("starting Mesos CNI server")
+	router := mux.NewRouter()
```

Comment:
cniLog() is a wrapper to netplugin log to append the plugin name at the end of each line for easy tracking